lib/plugins/aws/package/compile/events/alexaSkill/index.js

```diff
@@ -15,44 +15,74 @@ class AwsCompileAlexaSkillEvents {
   compileAlexaSkillEvents() {
     this.serverless.service.getAllFunctions().forEach((functionName) => {
       const functionObj = this.serverless.service.getFunction(functionName);
+      let alexaSkillNumberInFunction = 0;
 
       if (functionObj.events) {
         functionObj.events.forEach(event => {
+          let enabled = true;
+          let appId;
           if (event === 'alexaSkill') {
-            const lambdaLogicalId = this.provider.naming
-              .getLambdaLogicalId(functionName);
-
-            const permissionTemplate = {
-              Type: 'AWS::Lambda::Permission',
-              Properties: {
-                FunctionName: {
-                  'Fn::GetAtt': [
-                    lambdaLogicalId,
-                    'Arn',
-                  ],
-                },
-                Action: 'lambda:InvokeFunction',
-                Principal: 'alexa-appkit.amazon.com',
-              },
-            };
-
-            const lambdaPermissionLogicalId = this.provider.naming
-              .getLambdaAlexaSkillPermissionLogicalId(functionName);
-
-            const permissionCloudForamtionResource = {
-              [lambdaPermissionLogicalId]: permissionTemplate,
-            };
-
-            _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-              permissionCloudForamtionResource);
-          } else if (event.alexaSkill) {
+            const warningMessage = [
+              'Warning! You are using an old syntax for alexaSkill which doesn\'t',
+              ' restrict the invocation solely to your skill.',
+              ' Please refer to the documentation for additional information.',
+            ].join('');
+            this.serverless.cli.log(warningMessage);
+          } else if (typeof event.alexaSkill === 'string') {
+            appId = event.alexaSkill;
+          } else if (typeof event.alexaSkill === 'object') {
```

Comment:
Use `_.isPlainObject()` or `_.isObject()` here, dependeing if you'd allow inherited full objects or just plain objects.